v2/protocol/http/protocol_retry.go

```diff
@@ -42,7 +43,12 @@ func (p *Protocol) doOnce(req *http.Request) (binding.Message, protocol.Result)
 	if resp.StatusCode/100 == 2 {
 		result = protocol.ResultACK
 	} else {
-		result = protocol.ResultNACK
+		respbody, err := ioutil.ReadAll(resp.Body)
```

Comment:
Reading the body here will cause line 54 to error I think, because the body will have already been read. 

The body of the response is already returned within the message, this change should not be required.

Reply:
The change was made because the message is dropped later on here: https://github.com/cloudevents/sdk-go/blob/main/v2/protocol/http/protocol.go#L142

Reply:
How about we read and wrap the returned message from Request in Send and return it if err is not nil on L142? So fix the problem one bump up, rather than in protocol_retry. Seems like protocol_retry is doing the right thing, we just need to not drop the body if there is an error in Send.